vavr-jcstress/src/main/java/io/vavr/MemoizedConcurrencyTest.java

```diff
@@ -0,0 +1,415 @@
+package io.vavr;
+
+import org.openjdk.jcstress.annotations.*;
+import org.openjdk.jcstress.infra.results.LongResult1;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.FORBIDDEN;
+
+@Outcome(id = "1", expect = ACCEPTABLE, desc = "Source touched only once.")
+@Outcome(id = "2", expect = FORBIDDEN, desc = "Source touched twice.")
+public class MemoizedConcurrencyTest {
```

Comment:
Really good job @valery1707!
@danieldietrich, should these be added to the code generator?

Reply:
@paplorinc Good remark! Generally I would say yes but for now we can leave it as is. When we have the new inline code generator we are able to write self-expanding Java files (the generator code will be part of a Java comment). I will then change all Vavr source files accordingly...